.github/workflows/packageAndPublish.yaml

```diff
@@ -0,0 +1,69 @@
+on: 
+  push:
+    branches:
+      - '**'
+  pull_request:
+  release:
+    types: [released]
+
+name: packageAndPublish
+
+jobs:
+  package:
+    name: Package Extension
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - name: Install Node.js
+      uses: actions/setup-node@v1
+      with:
+        node-version: 10.x
+    - run: npm install
+    - name: Package
+      run: npm run package
+    - name: Upload Artifact
+      uses: actions/upload-artifact@v2
+      with:
+        name: extension-artifact
+        path: ./*.vsix
+
+  publish:
+    needs: package
+    name: Publish Extension  
+    runs-on: ubuntu-latest
+    if: github.event_name == 'release'
```

Comment:
This condition makes the publish job only run if the workflow was triggered by a release